content/blog/tidyselect-1-2-0/index.Rmd

```diff
@@ -0,0 +1,158 @@
+
+[^1]: If you are wondering whether a particular argument supports selections, look in the function documentation. Arguments tagged with `<tidy-select>` implement the selection dialect. By contrast, arguments tagged with `<data-masking>` only allow to refer to data frame columns directly.
+
+In this post, we'll cover the most important [lifecycle changes](https://lifecycle.r-lib.org/articles/stages.html) in the selection syntax that tidyverse users should know about. You can see a full list of changes in the [release notes](https://tidyselect.r-lib.org/news/index.html#tidyselect-120). We'll start with a quick recap of what it means in practice for a feature to be deprecated or soft-deprecated.
+
+```{r setup, message = FALSE}
+library(tidyverse)
+```
+
+Note: With this release of tidyselect, some error messages will be suboptimal until dplyr 1.1.0 is released (planned in late October). We recommend waiting until then before updating tidyselect (though it's not a big deal if you have already updated).
+
+
+## About soft-deprecation
+
+Deprecation of features in tidyverse packages is handled by the lifecycle package. See https://www.tidyverse.org/blog/2021/02/lifecycle-1-0-0/ for an introduction.
+
+The main feature of lifecycle is to distinguish between two stages of deprecation and two usage modes, direct and indirect.
+
+- For script users, __direct usage__ is when you use a deprecated feature from the global environment. If the deprecated feature was used inside a package function that you are calling, it is considered __indirect usage__.
+
+- For package developers, the distinction between direct and indirect usages is made by testthat in unit tests. If a function in your package calls the feature, it is considered direct usage. If that's a function in another package that you are calling, it's indirect usage.
+
+These usage modes determine how verbose (and thus how annoying) the deprecation warnings are.
+
+- For __soft-deprecation__, indirect usage is always silent because we only want to alert people who are actually able to update code at this point.
+
+  Direct usage is one warning every 8 hours to avoid being too annoying during this transition period, so that you can continue to work with existing code, ignore the warnings, and update to the new patterns on your own time.
+
+- For __deprecation__, it's now really time to update the code.  and direct usage gives a warning every time so that deprecated features can no longer be ignored.
+
+  Indirect usage now also warns, but only one warning every 8 hours since indirect users are not in control of the code that uses the deprecated feature. The warning message automatically picks up the package URL where the usage was detected so that you can easily report the deprecation to the relevant maintainers.
```

Comment:
```suggestion
  Indirect usage will now also warn, but only once every 8 hours since indirect users can't fix the problem themselves. The warning message automatically picks up the package URL where the usage was detected so that you can easily report the deprecation to the relevant maintainers.
```